### PR TITLE
DM-26415: remove chained commands from pipetask2 and DM-26388: pipetask2 --butler-config should accept a repository URI

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from lsst.daf.butler.cli.utils import option_section, split_kv, unwrap
+import lsst.obs.base.cli.opt as obsBaseOpts
+import lsst.daf.butler.cli.opt as dafButlerOpts
+from . import options as ctrlMpExecOpts
+
+instrumentOptionHelp = ("Add an instrument which will be used to load config overrides when defining a "
+                        "pipeline. This must be the fully qualified class name.")
+
+
+class OptionGroup():
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:38]: Did you want to add base class inside parentheses?

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    """Base class for an option group decorator. Requires the option group
+    subclass to have a property called `decorator`."""
+
+    def __call__(self, f):
+        for decorator in reversed(self.decorators):
+            f = decorator(f)
+        return f
+
+
+class pipeline_build_options(OptionGroup):  # noqa: N801
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:48]: Short one-line docstring could be useful for these classes.

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+class qgraph_options(OptionGroup):  # noqa: N801
+
+    def __init__(self):
+        self.decorators = [option_section(sectionText="Quantum graph building options:"),
+                           ctrlMpExecOpts.qgraph_option(),
+                           ctrlMpExecOpts.skip_existing_option(),
+                           ctrlMpExecOpts.save_qgraph_option(),
+                           ctrlMpExecOpts.save_single_quanta_option(),
+                           ctrlMpExecOpts.qgraph_dot_option()]
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:75]: Nitpick - there is a mixture of wrapping/indentation styles in different classes like

    self.decorators = [decorator1,
                       decorator2]

vs.

    self.decorators = [
        decorator1,
        decorator2
    ]

I personally prefer latter style, feels more readable as it lets you pack longer lines without more wrapping.

File: python/lsst/ctrl/mpexec/cli/script/build.py [hunk 4]
```diff
@@ -50,6 +52,10 @@ def build(order_pipeline=None, pipeline=None, pipeline_actions=(), pipeline_dot=
         Path location for storing resulting pipeline definition in YAML format.
     show : `list` [`str`]
         Descriptions of what to dump to stdout.
+    kwargs : `dict` [`str`, `str`]
+        Ignored; click commands may accept options for more than one script
+        function and pass all the option kwargs to each of the script functions
+        which ingore these unused kwargs.
```
Comment (andy-slac) [inline python/lsst/ctrl/mpexec/cli/script/build.py:58]: Hmm, this is potentially problematic, if for example client misspells option/keyword name, you won't catch it here. Could we do something safer here (and other places like this)?

Comment (n8pease) [reply python/lsst/ctrl/mpexec/cli/script/build.py:58]: @andy-slac I think there is not a good reason for the explicit kwargs to have default values. What if I removed them? That way if any of them are misspelled, and therefore missing, there will be an error.

Comment (n8pease) [reply python/lsst/ctrl/mpexec/cli/script/build.py:58]: (Click generates default value for all the options, the command function gets called with a kwarg for each of the `@option`s it is decorated with.

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py (hunk 1 above)
Comment (n8pease) [reply python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:38]: No, but I guess I can remove the parens?

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py (hunk 3 above)
Comment (n8pease) [reply python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:75]: Oh, latter is nice. 👍

File: python/lsst/ctrl/mpexec/cli/script/build.py (hunk 4 above)
Comment (andy-slac) [reply python/lsst/ctrl/mpexec/cli/script/build.py:58]: Having default values here and in click indeed bothers me too. If we say that all arguments are required than indeed we can drop defaults here and that should reduce the probability for spelling errors.

File: python/lsst/ctrl/mpexec/cli/opt/optionGroups.py (hunk 1 above)
Comment (andy-slac) [reply python/lsst/ctrl/mpexec/cli/opt/optionGroups.py:38]: 👍

**Pull request merged**